### ISSUE TITLE
Component Files not watched in nested pages

### ISSUE THREAD

**Issue by uptownhr** [nuxt/nuxt.js#2046]
When a component is imported on a multi-nested dynamic page, change the contents of the component is not detected and does not trigger a HMR re-build. 

STR
1. Create page structure
```
pages
- test
- - _x
- - - _y
- - - - index.vue
```
2. create a test component
3. load the test component from _y/index.vue
4. Change the contents of test component, notice a file watch change does not occur. 

Environment
Doceker node:8.8.0-alpine

Additional Notes.
This bug does not occur when the double nested dynamic directories do not exist. For example, change the `_y` to y will not experience the same problem.

```
pages
- test
- - _x
- - - y
- - - - index.vue
```

**Comment by clarkdo** [comment]
I have tried as your steps, it works for me 😄 . If I missed anything, pls let me know.

![kapture 2017-11-08 at 11 14 46](https://user-images.githubusercontent.com/4312154/32529889-b334e0e8-c400-11e7-8c85-b185b1caca88.gif)

**Comment by uptownhr** [comment]
it doesn't look like you missed anything. Let me try on a fresh project and see if I can also reproduce.

**Comment by uptownhr** [comment]
Alright, I have further investigated this issue and I believe it is related to ubuntu's default max watch limit. related to getting ENOSPC errors. I've tried reproducing the issue on a fresh project, inside and outside docker and could not reproduce. It was only reproduceable when I had all my docker containers running with 4 nuxt projects running on dev mode. As soon as I killed a container, the issue disappears.
